compiler/qsc_data_structures/src/language_features.rs

```diff
@@ -33,6 +35,8 @@ where
         iter.into_iter().fold(LanguageFeatures::empty(), |acc, x| {
             acc | match x.as_ref() {
                 "v2-preview-syntax" => LanguageFeatures::V2PreviewSyntax,
+                "preview-qir-gen" => LanguageFeatures::PreviewQirGen,
+                "adaptive-profile" => LanguageFeatures::AdaptiveProfile,
```

Comment:
Nothing seems to actually check this value.

Which actually kind of makes sense to me.... because doesn't the ability to pass `TargetProfile::Adaptive` imply that adaptive profile is supported? Is there ever a case where you'd want to pass in `TargetProfile::Adaptive` but NOT have the compiler respect it?